Source/WebCore/platform/network/curl/NetworkStorageSessionCurl.cpp

```diff
@@ -53,12 +53,12 @@ static String defaultCookieJarPath()
 #endif
 }
 
-static String cookiesForSession(const NetworkStorageSession& session, const URL& firstParty, const URL& url, bool forHTTPHeader)
+static String cookiesForSession(const NetworkStorageSession& session, const URL& firstParty, const URL& url, bool forHTTPHeader, IncludeSecureCookies includeSecureCookies)
 {
     StringBuilder cookies;
 
     auto searchHTTPOnly = forHTTPHeader ? std::nullopt : std::optional<bool> { false };
-    auto secure = url.protocolIs("https"_s) ? std::nullopt : std::optional<bool> { false };
+    auto secure = (includeSecureCookies == IncludeSecureCookies::Yes || url.protocolIs("https"_s)) ? std::nullopt : std::optional<bool> { false };
```

Comment:
I'm not an expert. But, looking at the code of WebCore and other ports, I think we should remove `|| url.protocolIs("https"_s)`.
https://github.com/WebKit/WebKit/blob/14d3cca9ff6908a009167387b63846c5db002d74/Source/WebCore/loader/CookieJar.cpp#L59